server/src/main/java/io/crate/expression/predicate/NotPredicate.java

```diff
@@ -206,7 +206,7 @@ public Query toQuery(Function input, LuceneQueryBuilder.Context context) {
         arg.accept(INNER_VISITOR, ctx);
         for (Reference reference : ctx.references()) {
             if (reference.isNullable()) {
-                builder.add(IsNullPredicate.refExistsQuery(reference, context), BooleanClause.Occur.MUST);
+                builder.add(IsNullPredicate.refExistsQuery(reference, context, false), BooleanClause.Occur.MUST);
```

Comment:
`False` here as below we check  `if (ctx.hasStrictThreeValuedLogicFunction)`  and add `genericFunctionQuery` with MUST if needed.